File: app/src/lib/models/project/ProjectDTO.ts
import type { UserDtoSimple } from '../user/UserDtoSimple';
import type { Language } from './Language';

type ProjectDTO = {
	id: string;
	name: string;
	description: string;
	license: string;
	likes: number;
	liked: boolean;
	created: string;
	codeSize: string;
	owner: UserDtoSimple;
	creators: UserDtoSimple[];
	private: boolean;
	latestReleaseVersion: string;
	languages: Language[];
	tags: string[];
};

export type { ProjectDTO };

File: server/src/main/java/com/example/projectstore/project/ProjectDTO.java
package com.example.projectstore.project;

import com.example.projectstore.user.UserDtoSimple;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProjectDTO {
    private UUID id;
    private String name;
    private String description;
    private String license;
    private int likes;
    private Boolean liked;
    private String codeSize;
    private String mainLanguage;
    private LocalDateTime created;
    private UserDtoSimple owner;
    private List<UserDtoSimple> creators;
    private boolean isPrivate;
    private String latestReleaseVersion;
    List<LanguageDto> languages;
    Set<String> tags;
}
